src/agent/src/initdata.rs

```diff
@@ -0,0 +1,636 @@
+
+        let initdata_digest = base64::engine::general_purpose::STANDARD.encode(digest);
+        (initdata_digest, initdata.data)
+    } else {
+        debug!(
+            sl!(),
+            "SetInitdata called with empty initdata. Launch guest components with default configs"
+        );
+        (DEFAULT_INITDATA_DIGEST.into(), DefinedFields::default())
+    };
+
+    init_attestation_components(
+        &sl!(),
+        configs.aa_config,
+        configs.cdh_config,
+        initdata_digest,
+    )
+    .await
+    .context("launch guest components failed")?;
+
+    if let Some(_policy) = configs.policy {
+        debug!(sl!(), "Agent policy is given, try to set policy.");
+
+        #[cfg(feature = "agent-policy")]
+        {
+            let mut policy_agent = crate::AGENT_POLICY.lock().await;
+            // SetInitData will allow a one-time SetPolicy. By default policy allow to set, thus we
+            // do not need to check if API is allowed.
+            policy_agent
+                .set_policy(&_policy)
+                .await
+                .context("set policy failed")?;
+
+            debug!(sl!(), "Set policy successfully.");
+        }
+
+        #[cfg(not(feature = "agent-policy"))]
+        debug!(sl!(), "Feature `agent-policy` is not enabled in kata-agent, ignore the policy field in initdata");
+    }
+
+    Ok(())
+}
+
+async fn wait_for_path_to_exist(logger: &Logger, path: &str, timeout_secs: i32) -> Result<()> {
+    let p = Path::new(path);
+    let mut attempts = 0;
+    loop {
+        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+        if p.exists() {
+            return Ok(());
+        }
+        if attempts >= timeout_secs {
+            break;
+        }
+        attempts += 1;
+        info!(
+            logger,
+            "waiting for {} to exist (attempts={})", path, attempts
+        );
+    }
+
+    Err(anyhow!("wait for {} to exist timeout.", path))
+}
+
+pub async fn launch_process(
+    logger: &Logger,
+    path: &str,
+    args: Vec<String>,
+    unix_socket_path: &str,
+    timeout_secs: i32,
+) -> Result<()> {
+    if !Path::new(path).exists() {
+        bail!("path {} does not exist.", path);
+    }
+    if !unix_socket_path.is_empty() && Path::new(unix_socket_path).exists() {
+        fs::remove_file(unix_socket_path).await?;
+    }
+    Command::new(path).args(args).spawn()?;
+    if !unix_socket_path.is_empty() && timeout_secs > 0 {
+        wait_for_path_to_exist(logger, unix_socket_path, timeout_secs).await?;
+    }
+
+    Ok(())
+}
+
+async fn launch_confidential_data_hub(cdh_config: Option<String>) -> Result<()> {
+    if Path::new(CDH_PATH).exists() {
+        let args = match cdh_config {
+            Some(config) => {
+                debug!(
+                    sl!(),
+                    "Launch Confidential Data Hub with config delivered via Initdata"
+                );
+                tokio::fs::write(CDH_CONFIG_PATH, config.as_bytes()).await?;
```

Comment:
Can we move the logic of writing an CDH config file out of the lauch-process fn? We will also need to write a config file if we launch CDH via systemd.